src/library/scala/collection/concurrent/Map.scala

```diff
@@ -131,4 +131,22 @@ trait Map[K, V] extends scala.collection.mutable.Map[K, V] {
       case _ => this.updateWithAux(key)(remappingFunction)
     }
   }
+
+  private[collection] def filterInPlaceImpl(p: (K, V) => Boolean): this.type = {
+    val it = iterator
+    while (it.hasNext) {
+      val (k, v) = it.next()
+      if (p(k, v)) remove(k, v)
```

Comment:
oops I inverted the condition by accident

```suggestion
      if (!p(k, v)) remove(k, v)
```

Reply:
fixed in #9737